superset/db_engine_specs.py

```diff
@@ -74,6 +81,103 @@ def extra_table_metadata(cls, database, table_name, schema_name):
         """Returns engine-specific table metadata"""
         return {}
 
+    @staticmethod
+    def csv_to_df(names, filepath_or_buffer, sep, header, index_col, squeeze,
+                  prefix, mangle_dupe_cols, skipinitialspace, skiprows, nrows,
+                  skip_blank_lines, parse_dates, infer_datetime_format,
+                  dayfirst, thousands, decimal, quotechar, escapechar, comment,
+                  error_bad_lines, chunksize):
+        # Use Pandas to parse csv file to a dataframe
+        upload_path = config['UPLOAD_FOLDER'] + filepath_or_buffer
+        # Expose this to api so can specify each field
+        chunks = pandas.read_csv(filepath_or_buffer=upload_path,
+                                 sep=sep,
+                                 header=header,
+                                 names=names,
+                                 index_col=index_col,
+                                 squeeze=squeeze,
+                                 prefix=prefix,
+                                 mangle_dupe_cols=mangle_dupe_cols,
+                                 skipinitialspace=skipinitialspace,
+                                 skiprows=skiprows,
+                                 nrows=nrows,
+                                 skip_blank_lines=skip_blank_lines,
+                                 parse_dates=parse_dates,
+                                 infer_datetime_format=infer_datetime_format,
+                                 dayfirst=dayfirst,
+                                 thousands=thousands,
+                                 decimal=decimal,
+                                 quotechar=quotechar,
+                                 escapechar=escapechar,
+                                 comment=comment,
+                                 encoding='utf-8',
+                                 error_bad_lines=error_bad_lines,
+                                 chunksize=chunksize,
+                                 iterator=True)
+        df = pandas.DataFrame()
+        df = pandas.concat(chunk for chunk in chunks)
+        print(df)
+        return df
+
+    @staticmethod
+    def df_to_db(df, name, con, schema, if_exists, index,
+                 index_label, chunksize):
+
+        engine = create_engine(con, echo=False)
+
+        # Use Pandas to parse dataframe to database
+        df.to_sql(name=name, con=engine, schema=schema, if_exists=if_exists,
+                  index=index, index_label=index_label, chunksize=chunksize)
+
+
+    @staticmethod
+    def upload_csv(form):
+        print(results_backend)
+        def allowed_file(filename):
+            # Only allow specific file extensions as specified in the config
+            return '.' in filename and \
```

Comment:
using index `[-1]` would allow for filenames with multiples dots in them, but better is to use the `os.path` module as in https://docs.python.org/2/library/os.path.html#os.path.splitext